src/test/java/org/csanchez/jenkins/plugins/kubernetes/pipeline/ContainerExecDecoratorPipelineTest.java

```diff
@@ -121,15 +121,32 @@ public void containerEnvironmentIsHonored() throws Exception {
         assertNotNull(createJobThenScheduleRun());
         r.waitForCompletion(b);
         r.assertLogContains(
-                "from Groovy: /opt/java/openjdk/bin:/usr/local/sbin:/usr/local/bin:/usr/sbin:/usr/bin:/sbin:/bin", b);
+                "from Groovy outside container: /opt/java/openjdk/bin:/usr/local/sbin:/usr/local/bin:/usr/sbin:/usr/bin:/sbin:/bin",
+                b);
+        r.assertLogContains(
+                "from shell outside container: /opt/java/openjdk/bin:/usr/local/sbin:/usr/local/bin:/usr/sbin:/usr/bin:/sbin:/bin",
+                b);
         r.assertLogContains(
-                "outside container: /opt/java/openjdk/bin:/usr/local/sbin:/usr/local/bin:/usr/sbin:/usr/bin:/sbin:/bin",
+                "from Groovy outside container with override: /bar:/opt/java/openjdk/bin:/usr/local/sbin:/usr/local/bin:/usr/sbin:/usr/bin:/sbin:/bin",
                 b);
         r.assertLogContains(
-                "outside container with override: /bar:/opt/java/openjdk/bin:/usr/local/sbin:/usr/local/bin:/usr/sbin:/usr/bin:/sbin:/bin",
+                "from shell outside container with override: /bar:/opt/java/openjdk/bin:/usr/local/sbin:/usr/local/bin:/usr/sbin:/usr/bin:/sbin:/bin",
                 b);
-        r.assertLogContains("inside container: /usr/local/sbin:/usr/local/bin:/usr/sbin:/usr/bin:/sbin:/bin", b);
+        // When using groovy, the environment relies on the computer's environment.
         r.assertLogContains(
-                "inside container with override: /bar:/usr/local/sbin:/usr/local/bin:/usr/sbin:/usr/bin:/sbin:/bin", b);
+                "from Groovy inside container: /opt/java/openjdk/bin:/usr/local/sbin:/usr/local/bin:/usr/sbin:/usr/bin:/sbin:/bin",
+                b);
+        r.assertLogContains(
+                "from shell inside container: /usr/local/sbin:/usr/local/bin:/usr/sbin:/usr/bin:/sbin:/bin", b);
+        // When using groovy, the environment relies on the computer's environment, using container step doesn't change
+        // anything.
+        r.assertLogContains(
+                "from Groovy inside container with override: /bar:/opt/java/openjdk/bin:/usr/local/sbin:/usr/local/bin:/usr/sbin:/usr/bin:/sbin:/bin",
```

Comment:
My point in https://github.com/jenkinsci/kubernetes-plugin/pull/1486#discussion_r1432892359 & https://github.com/jenkinsci/kubernetes-plugin/pull/1486#issuecomment-1864729718 is that here we should actually expect `/bar:/usr/local/sbin:…` just like from the `sh` step, even if we do not consider it _important_ to fix.